### PR TITLE
Generate Line Count Tables

### PR DESCRIPTION
**Pull request by ryan-berger**

### CODE REVIEW

**Comment by ryan-berger** [comment]
This may want to run as part of CI as the dependency upon `scc` and `jq` makes this less portable than I would like @oflatt

File: writing/gen-linecount.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    total_lines = 0
+    file_paths = glob.glob("../../**/*.egg", recursive=True)
+    for file_path in file_paths:
+        with open(file_path, 'r') as file:
+            lines = file.readlines()
+            total_lines += len(lines)
+    return total_lines
+
+def get_rust_lines():
+    rust_lines_output = subprocess.run(["scc", "--format", "json", "../../"], capture_output=True)
```
[the remark this answers is not in the document]
Comment (ryan-berger) [reply writing/gen-linecount.py:22]: We can't use `loc` due to limitations on its output formats (it only outputs ASCII art that would be nearly impossible to parse) and it is no longer maintained, so the author recommended `scc`.

However, there is another line counting tool called `tokei` which also is in Rust and looks like it will do what we want. I will switch this over to tokei

File: writing/tmpls/linecount.tex [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+\begin{tabular}{ |s|p{2cm}| }
```
[the remark this answers is not in the document]
Comment (ryan-berger) [reply writing/tmpls/linecount.tex:2]: Hmmmm, I'm unsure what we get out of a generic json -> latex converter if we want to do anything custom in the _slightest_ with the table (meaning not this exact piece of text). Having this turn into anything more than that and we get into crazy territory where we're reinventing the wheel and should get a text templating tool involved.

Also, this is pretty typical string manipulation for templating language. Python doesn't have text templating built into its std lib (like Go does for example) so unless we also want people to set up a Python environment for something like jinja/Django's templates, I vote for hacky find and replace (for this table in particular)

An alternative would be to keep the string as a format string _in Python_ as a multiline, and then generate it on command.

TLDR, my opinion is we either:
a) build the string (how the nightly table will get done, how we _could_ do this one)
b) find and replace inside the string ("hacky", but also how lots of text templating works in the real world)

File: writing/linecount.tex [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+\begin{tabular}{ |s|p{2cm}| }
```
[the remark this answers is not in the document]
Comment (ryan-berger) [reply writing/linecount.tex:2]: I honestly think this probably should be committed. I see no reason as to why it is a missing file and needs to be regenerated on every computer.

It may want to get generated on every push to main, but I don't now if I see a reason to ignore it. I can if you'd like though, and then when we import it from `main.tex` it will fail to compile if you haven't generated the table yet

File: writing/tmpls/linecount.tex (hunk 2 above)
[the remark this answers is not in the document]
Comment (oflatt) [reply writing/tmpls/linecount.tex:2]: Okay, I think the "build the string" option is better

File: writing/linecount.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (oflatt) [reply writing/linecount.tex:2]: Any files that can be generated should be generated
"push button eval" philosophy that has served me well in the past

File: writing/tmpls/linecount.tex (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryan-berger) [reply writing/tmpls/linecount.tex]: Done

File: writing/linecount.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (oflatt) [reply writing/linecount.tex:2]: I'm going to insist we don't commit this file

File: writing/gen-linecount.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        with open(file_path, 'r') as file:
+            lines = file.readlines()
+            total_lines += len(lines)
+    return total_lines
+
+def get_rust_lines():
+    rust_lines_output = subprocess.run(["tokei", "--output", "json", "../../"], capture_output=True)
+    return json.loads(rust_lines_output.stdout)["Rust"]["code"]
+
+def replace_in_text(file_path, replacements):
```
Comment (oflatt) [inline writing/gen-linecount.py:24]: unused method

**Review by oflatt — CHANGES_REQUESTED** [review]
Some small changes
Don't forget to add this to the nightly in another PR!

File: writing/linecount.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryan-berger) [reply writing/linecount.tex:2]: I was very confused on what you meant by "push button eval", I thought that'd mean keep it in.

I'm very happy to get rid of the file, lets keep a .gitignore inside this directory for any generated latex files

**Review by oflatt — APPROVED** [review]
Looks good, you'll want to remove the stray gitignore change
